### ISSUE TITLE
LARC optimizer returns incorrect param groups after loading from checkpoints

### ISSUE THREAD

**Issue by mannatsingh** [NVIDIA/apex#766]
Upon resuming from a checkpoint, users need to call `larc_optim.load_state_dict(state_dict)`, where the state is fetched by calling `state_dict = larc_optim.state_dict()`. This however breaks `larc_optim.param_groups` - after the call, `larc_optim.param_groups` refers to a different dictionary from the one returned by `larc_optim.optim.param_groups`. 

This happens because `larc_optim.param_groups` is initialized to `larc_optim.optim.param_groups` during `__init__()` (https://github.com/NVIDIA/apex/blob/master/apex/parallel/LARC.py#L40), but then `larc_optim.optim.load_state_dict(state_dict)` makes `larc_optim.optim.param_groups` point to a new dictionary (https://github.com/pytorch/pytorch/blob/master/torch/optim/optimizer.py#L157).

This can lead to bugs when the code relies on `larc_optim.param_groups`. For instance, learning rate schedulers use `optim.param_groups` (https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L36), but this has no effect on the underlying optimizer.
